### PR TITLE
Fog Rewrite and Mixin changes

### PR DESCRIPTION
**Pull request by JustAlittleWolf**
Changed fog removal to also work with sodium
Changed server check so that the server doesn't crash

Changed a lot of @Overwrite calls to @Inject for improved simplicity and code readability

### CODE REVIEW

**Comment by JustAlittleWolf** [comment]
Also do you have a discord?

**Comment by AnOpenSauceDev** [comment]
This a pretty good PR!
 I'll probably merge it when I can.

> Also do you have a discord?

I don't really use discord to be honest.

File: src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java [hunk 1]
```diff
@@ -1,151 +1,52 @@
...
-            }
-
-            red += (s - red) * r;
-            green += (t - green) * r;
-            blue += (u - blue) * r;
-
-
+    @Inject(method = "applyFog", at = @At("HEAD"), cancellable = true)
+    private static void applyFog(Camera camera, BackgroundRenderer.FogType fogType, float viewDistance, boolean thickFog, float tickDelta, CallbackInfo ci) {
+        if (!Methane.ModActive) return;
```
Comment (AnOpenSauceDev) [inline src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java:24]: not so sure if this check needs to be here-  I think the changes might be better if they persist.

**Review by AnOpenSauceDev — APPROVED** [review]

File: src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java (hunk 1 above)
Comment (JustAlittleWolf) [reply src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java:24]: yes you could do that, but then the "toggle methane" key would only toggle the fullbright functionality, which is not the entire mod

Comment (AnOpenSauceDev) [reply src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java:24]: I think this calls for a config option in that case.

Comment (AnOpenSauceDev) [reply src/main/java/me/wolfie/methane/mixin/BackgroundMixin.java:24]: https://github.com/AnOpenSauceDev/Methane-mod/commit/b998c5c435c80f47f61e7bdbf9c037b58677bab5 kinda fixes this issue

**Pull request merged**